app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -103,7 +103,17 @@ export default Controller.extend(bufferedProperty("model"), {
   ),
 
   updateQueryParams() {
-    this.setProperties(this.get("model.postStream.streamFilters"));
+    const filters = this.get("model.postStream.streamFilters");
+
+    if (Object.keys(filters).length > 0) {
+      this.setProperties(filters);
```

Comment:
Before this fix, the properties were not updated because `this.get("model.postStream.streamFilters")` was returning `{}`